pkg/flare/manifests.go

```diff
@@ -0,0 +1,206 @@
+	var b bytes.Buffer
+
+	deploy, err := cl.Cl.AppsV1().Deployments(namespace).Get(context.TODO(), name, metav1.GetOptions{})
+	if err != nil {
+		log.Debugf("Can't retrieve Deployment %v from the API server: %s", name, err.Error())
+		return nil, err
+	}
+
+	deployjson, err := json.Marshal(deploy)
+	deployyaml, err := yaml.JSONToYAML(deployjson)
+	fmt.Fprintln(&b, string(deployyaml))
+	return b.Bytes(), nil
+}
+
+// getDeployedHelmConfigmap returns the configmap for a given release.
+// Only a single release for a given name can deployed at one time.
+func getDeployedHelmConfigmap(cl *apiserver.APIClient, name string, namespace string) (*v1.ConfigMap, error) {
+	var selector string
+
+	selector = labels.Set{
+		"owner":  "helm",
+		"status": "deployed",
+		"name":   name,
+	}.AsSelector().String()
+	configmapList, err := cl.Cl.CoreV1().ConfigMaps(namespace).List(context.TODO(), metav1.ListOptions{LabelSelector: selector})
+	if err != nil {
+		return nil, err
+	}
+	if len(configmapList.Items) != 1 {
+		return nil, log.Errorf("%s configmaps found, but expected 1", fmt.Sprint((configmapList.Items)))
+	}
+	return &configmapList.Items[0], nil
+}
+
+// getDeployedHelmSecret returns the secret for a given release.
+// Only a single release for a given name can deployed at one time.
+func getDeployedHelmSecret(cl *apiserver.APIClient, name string, namespace string) (*v1.Secret, error) {
+	var selector string
+
+	selector = labels.Set{
+		"owner":  "helm",
+		"status": "deployed",
+		"name":   name,
+	}.AsSelector().String()
+	secretList, err := cl.Cl.CoreV1().Secrets(namespace).List(context.TODO(), metav1.ListOptions{LabelSelector: selector})
+	if err != nil {
+		return nil, err
+	}
+	if len(secretList.Items) != 1 {
+		return nil, log.Errorf("%s secrets found, but expected 1", fmt.Sprint(len(secretList.Items)))
+	}
+	return &secretList.Items[0], nil
+}
+
+// Retrieve Helm chart user values from the API server for a given name and namespace looking through Secrets or Configmaps
+func GetDatadogHelmUserValues(cl *apiserver.APIClient, name string, namespace string, storage string) ([]byte, error) {
+	switch storage {
+	case "secret":
+		secret, err := getDeployedHelmSecret(cl, name, namespace)
+		if err != nil {
+			return nil, err
+		}
+		b64EncodedGzipRelease := secret.Data["release"]
+		// Contrary to the Configmap, the secret data is a byte array, so the string function is necessary
+		b, err := decodeChartValuesFromRelease(string(b64EncodedGzipRelease))
+		if err != nil {
+			return nil, err
+		}
+		return b, nil
+	case "configmap":
+		configmap, err := getDeployedHelmConfigmap(cl, name, namespace)
+		if err != nil {
+			return nil, err
+		}
+		b64EncodedGzipRelease := configmap.Data["release"]
+		b, err := decodeChartValuesFromRelease(b64EncodedGzipRelease)
+		if err != nil {
+			return nil, err
+		}
+		return b, nil
+	default:
+		return nil, log.Errorf("The requested storage %s is not supported", storage)
+	}
+}
```

Comment:
you can factorized a bit this function

```suggestion
func GetDatadogHelmUserValues(cl *apiserver.APIClient, name string, namespace string, storage string) ([]byte, error) {
	var dataString string
	switch storage {
	case "secret":
		secret, err := getDeployedHelmSecret(cl, name, namespace)
		if err != nil {
			return nil, err
		}
		// Contrary to the Configmap, the secret data is a byte array, so the string function is necessary
		dataString := string(secret.Data["release"])	
	case "configmap":
		configmap, err := getDeployedHelmConfigmap(cl, name, namespace)
		if err != nil {
			return nil, err
		}
		dataString = configmap.Data["release"]
	default:
		return nil, log.Errorf("The requested storage %s is not supported", storage)
	}
	return decodeChartValuesFromRelease(dataString)
}
```